kartothek/io/eager.py

```diff
@@ -482,7 +482,10 @@ def store_dataframes_as_dataset(
     if dataset_uuid is None:
         dataset_uuid = gen_uuid()
 
-    if isinstance(dfs, dict):
+    if isinstance(dfs, list) and isinstance(dfs[0], tuple):
```

Comment:
I'd prefer this logic to be implemented in `parse_input_to_metapartition` since this is used for all interfaces. You can also document the new input format there